superset/db_engine_specs.py

```diff
@@ -1485,20 +1503,21 @@ class PinotEngineSpec(BaseEngineSpec):
     inner_joins = False
     supports_column_aliases = False
 
-    _time_grain_to_datetimeconvert = {
+    # Pinot does its own conversion below
+    time_grain_functions: Dict[Optional[str], str] = {
```

Comment:
My concern is renaming this to time_grain_functions is that they are not really behaving like "functions" (like in the rest of the db engine specs). 

For example if someone (other than the pinot specific get_timestamp_expr) starts using them, then they wouldn't be getting a function. So I would be wary of changing this. 

If anything, we can make time_grain_functions = None (in case they are set in the BaseEngineSpec), so that no one can use them. 

And switch this back to its internal name that the below pinot specific get_timestamp_expr can use.

Reply:
The way I see it all values in `time_grain_functions` are merely technical strings that are used by the spec to construct the final TimestampExpression. The fact that Pinot does this slightly differently I think is ok; all interaction with them should always be done via the `get_timestamp_expr()` function anyway, be it Pinot or Postgres. But I agree that the naming is slightly confusing. I would personally stick with my proposal, but I will change it back if you feel it is more intuitive.

Reply:
Okay. This was a nit anyway :D.

Should we make time_grain_functions private then ? _time_grain_functions ? 

The name is a bit of a misnomer now: The values in this dictionary are no longer pure functions.

Or we can just punt on this and perhaps the comment above "time_grain_functions are merely technical strings ...." as a code comment.

Its fine either way

Reply:
I agree, it should be made private as it is. However, I think I'll let this be as it is and keep this in mind next time this time grains are refactored (originally they were a tuple of tuples which are currently emulated via `BaseEngineSpec.get_time_grains()`). Perhaps next iteration they'll become more final.